sources/packages/backend/libs/integrations/src/services/disbursement-schedule/e-cert-processing-steps/add-aviation-credential-restriction-step.ts

```diff
@@ -0,0 +1,92 @@
+import { Injectable } from "@nestjs/common";
+import { EntityManager } from "typeorm";
+import {
+  RestrictionCode,
+  StudentRestrictionSharedService,
+  SystemUsersService,
+} from "@sims/services";
+import { StudentRestriction } from "@sims/sims-db";
+import { ECertProcessStep } from "./e-cert-steps-models";
+import { ProcessSummary } from "@sims/utilities/logger";
+import { EligibleECertDisbursement } from "../disbursement-schedule.models";
+import { ECertGenerationService } from "../e-cert-generation.service";
+import { getRestrictionByCode } from "./e-cert-steps-utils";
+import { RestrictionService } from "@sims/integrations/services";
+
+/**
+ * Check if the offering belongs to aviation credential types and add an aviation restriction
+ * for the given credential type if one does not already exist.
+ * By adding this restriction, it will prevent the student from being funded for the same
+ * aviation credential type again in the future.
```

Comment:
Can we add an extra comment as
> Aviation offerings are not expected to create second disbursements due to their number of funded weeks. In case this scanrios changes, the second disbursement would be blocked if no changes are made.